File: src/Marsen.NetCore.Dojo/Classes/Joey/Tennis/StateV2/AdvState.cs
namespace Marsen.NetCore.Dojo.Classes.Joey.Tennis.StateV2;

public class AdvState : State
{
    public override string Score()
    {
        return $"{Winner()} Adv";
    }

    protected override void ChangeState()
    {
        if (IsSame())
            Game.ChangeState(new DeuceState());
        else
            Game.ChangeState(new WinState());
    }
}

File: src/Marsen.TypeScript.Dojo/src/Kata/AdvState.ts
import { WinState } from "./WinState";
import { State } from "./State";
import { DeuceState } from "./DeuceState";
export class AdvState extends State {
    Score(): string {
        return `${this.Winner()} Adv`;
    }

    ChangeState(): void {
        if (this.IsSame()) {
            this.Context.State = new DeuceState();
        }
        else {
            this.Context.State = new WinState();
        }
    }
}
